framework/SwiftPlot/FloatConvertible.swift

```diff
@@ -0,0 +1,25 @@
+public protocol FloatConvertible {
+    init(_ other: Float)
+    init(_ other: Double)
+    // init(_ other: Int)
+    init(_ x: FloatConvertible)
+
+    func _asOther<T:FloatConvertible>() -> T
```

Comment:
This protocol seems overly complicated in its construction. What you really want is to use this to be able to convert other types to Float. You want it to be as simple as Float(value) producing a Float.

For that, you don't need the initializers here. Instead, I'd define an extension on Float that provides an init like the following:

```
extension Float {
	    public init(_ x:FloatConvertible) {self = x.toFloat()}
}
```

and I'd remove the other initializers here. You're not directly initializing a FloatConvertible type, you simply want to make sure that you can filter for types that can be used with Float(value).

I'm using `toFloat()` as a method name here, because you don't want to be using an underscore for a public method name. The signature should be something like the following:

```
func toFloat() -> Float
```

Reply:
> You're not directly initializing a FloatConvertible type, you simply want to make sure that you can filter for types that can be used with Float(value).

@BradLarson for the LineChart example I've specified below, if I have to declare  
Axis<FloatConvertible,FloatConvertible> I'll need to initialize a FloatConvertible type. And therefore we'll need those initializers. Otherwise I get this error:
`error: 'FloatConvertible' is not convertible to 'Float'`

Reply:
@BradLarson according to the LineChart example I've given below I'll need to initialize a FloatConvertible...
So I'll need to write those initializers.